pkg/movingaverage/weight_moving_average.go

```diff
@@ -38,39 +40,36 @@ func NewWMA(sizes ...int) *WMA {
 
 // Add adds a data point.
 func (w *WMA) Add(n float64) {
+	w.score = w.score - w.sum + n*float64(w.size)
+	if w.count < w.size {
+		w.sum = w.sum + n
+	} else {
+		w.sum = w.sum - w.records[(w.count)%w.size] + n
+	}
 	w.records[w.count%w.size] = n
 	w.count++
 }
 
 // Get returns the weight average of the data set.
 func (w *WMA) Get() float64 {
 	if w.count == 0 {
-		return 0
+		return w.score / float64(w.size)
```

Comment:
At this time, the return result must be 0?

Reply:
yes， it should be 0 when the list has no element

Reply:
w. score is zero when w. count is zero? I think return directly 0 is enough